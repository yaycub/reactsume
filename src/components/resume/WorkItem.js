import React from 'react';
import PropTypes from 'prop-types';

const WorkItem = ({ business, titleWithDates, didWhat }) => {
  const didWhatEls = didWhat.map((item, i) => {
    return (
      <li key={i}>
        {item}
      </li>
    );
  });

  return (
    <section>
      <h4>{business}</h4>
      <p>{titleWithDates}</p>
      <ul>
        {didWhatEls}
      </ul>
    </section>
  );
};

WorkItem.propTypes = {
  business: PropTypes.string.isRequired,
  titleWithDates: PropTypes.string.isRequired,
  didWhat: PropTypes.array.isRequired
};

export default WorkItem;
