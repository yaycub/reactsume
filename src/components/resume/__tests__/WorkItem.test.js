import React from 'react';
import { shallow } from 'enzyme';
import data from '../../../resume.json';
import WorkItem from '../WorkItem.js';

describe('WorkItem Component', () => {
  it('can render a WorkItem', () => {
    const { work } = data;
    const wrapper = shallow(<WorkItem {...work[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
