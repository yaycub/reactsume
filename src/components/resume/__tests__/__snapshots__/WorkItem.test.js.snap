// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkItem Component can render a WorkItem 1`] = `
<section>
  <h4>
    Airbnb
  </h4>
  <p>
    Senior Payments Specialist | May 2014 - April 2019
  </p>
  <ul>
    <li
      key="0"
    >
      Identified impact of bugs using SQL queries and notified engineering via Jira
    </li>
    <li
      key="1"
    >
      Built a partnership between Airbnb Collections and Airbnb Payments CS
    </li>
    <li
      key="2"
    >
      Trained multiple entry level associates on the Payments CS team & partner CS
    </li>
    <li
      key="3"
    >
      Consistent top performer on the team
    </li>
  </ul>
</section>
`;
